File: src/Defender.BudgetTracker/src/Domain/Entities/DiagramSetup/RegularExpenseDiagramSetup.cs
using Defender.BudgetTracker.Domain.Entities.Interfaces;
using Defender.BudgetTracker.Domain.Enums;
using Defender.Common.Entities;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace Defender.BudgetTracker.Domain.Entities.DiagramSetup;

public class RegularExpenseDiagramSetup : IUserOwnedModel, IBaseModel
{
    private DateOnly? _endMonth;

    [BsonId]
    [BsonRepresentation(BsonType.String)]
    public Guid Id { get; set; }

    [BsonGuidRepresentation(GuidRepresentation.CSharpLegacy)]
    public Guid UserId { get; set; }

    [BsonRepresentation(BsonType.String)]
    public Currency MainCurrency { get; set; }

    public int? LastMonths { get; set; }

    public DateOnly? EndMonth
    {
        get => _endMonth;
        set => _endMonth = value.HasValue
            ? new DateOnly(value.Value.Year, value.Value.Month, 1)
            : null;
    }
}

File: src/Defender.Portal/src/WebUI/ClientApp/src/models/budgetTracker/regularExpenses/RegularExpenseDiagramSetup.ts
import { Currency } from "src/models/shared/Currency";

export interface RegularExpenseDiagramSetup {
  id?: string;
  userId?: string;
  mainCurrency: Currency;
  lastMonths?: number;
  endMonth?: string;
}

export interface RegularExpenseChartRow {
  month: string;
  totalComfort: number;
  regular: number;
  subscription: number;
  annual: number;
}
